src/infrastructure/user/mongo.repository.ts

```diff
@@ -0,0 +1,46 @@
+import { UserRepository } from '../../domain/user/user.repository';
+import { User } from '../../domain/user/user.entity';
+import UserModel from './user.model';
+import { UserUpdateProps } from '../../domain/user/user.value';
+import { CustomError, RestCodes } from '../../framework/errorFactory';
+
+export class MongoRepository implements UserRepository {
+    async createUser(user: Partial<User>): Promise<User | null> {
+        const newUser = new UserModel(user);
+        const savedUser = await newUser.save();
+        return savedUser.toObject();
+    }
+
+    async getUserById(userId: string): Promise<User | null> {
+        const user = await UserModel.findOne({ userId: userId }).lean();
+        return user || null;
+    }
+
+    async updateUser(
+        userId: string,
+        updatedData: UserUpdateProps,
+    ): Promise<User> {
+        const updatedUser = await UserModel.findOneAndUpdate(
+            { userId: userId },
+            { $set: updatedData },
+            { new: true },
+        ).lean();
+        if (!updatedUser) {
+            throw new CustomError(
```

Comment:
Here you can not throw rest errors. You should throw a Error with a message and catch it in upper layers. Remember, this implementatios it's on infrastructure but is used from domain layer.